internal/gopsutilenv/gopsutilenv_linux.go

```diff
@@ -0,0 +1,103 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build linux
+
+package gopsutilenv // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/gopsutilenv"
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/shirou/gopsutil/v4/common"
+)
+
+var gopsutilEnvVars = map[common.EnvKeyType]string{
+	common.HostProcEnvKey: "/proc",
+	common.HostSysEnvKey:  "/sys",
+	common.HostEtcEnvKey:  "/etc",
+	common.HostVarEnvKey:  "/var",
+	common.HostRunEnvKey:  "/run",
+	common.HostDevEnvKey:  "/dev",
+}
+
+// This exists to validate that different components that use this module do not
+// have inconsistent root_path configurations. The root_path is passed down to gopsutil
+// through context, so it must be consistent across the process.
+var globalRootPath string
+
+func ValidateRootPath(rootPath string) error {
+	if rootPath == "" || rootPath == "/" {
+		return nil
+	}
+
+	if globalRootPath != "" && rootPath != globalRootPath {
+		return fmt.Errorf("inconsistent root_path configuration detected among components: `%s` != `%s`", globalRootPath, rootPath)
+	}
+	globalRootPath = rootPath
+
+	if _, err := os.Stat(rootPath); err != nil {
+		return fmt.Errorf("invalid root_path: %w", err)
+	}
+
+	return nil
+}
+
+func SetGoPsutilEnvVars(rootPath string) (common.EnvMap, error) {
+	m := common.EnvMap{}
+	if rootPath == "" || rootPath == "/" {
+		return m, nil
+	}
+
+	if err := ValidateRootPath(rootPath); err != nil {
```

Comment:
Why do we need to add this here? The validation should be called on the config validation step. I'd say let's move the code as is

Reply:
I have noticed that users might end up forgetting to call it and my thought was proved in the integration test